recipes/sentry-native/all/conanfile.py

```diff
@@ -30,42 +30,47 @@ class SentryNativeConan(ConanFile):
         "transport": "curl"
     }
 
+    _cmake = None
+
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
-    _cmake = None
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+        if self.options.backend == "inproc" and self.settings.os == "Windows" and tools.Version(self.version) < "0.4":
+            raise ConanInvalidConfiguration("The in-process backend is not supported on Windows")
+        if self.options.transport == "winhttp" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("The winhttp transport is only supported on Windows")
```

Comment:
This should be in `validate(self)`

Reply:
It requires consumers to use conan 1.32.0, and there is no advantage to move this in `validate()` since those checks don't access dependencies options/versions. All values are available before dependency graph resolution under this recipe.

Reply:
a user could have a requirement which sets `self.options["sentry-native"].backend` to some value, and another requirement which sets `self.options["sentry-native"].backend` to another value

Reply:
That's not new, at `configure()` time of a recipe, its own final options values are known (but not its own dependencies options values or even versions).

Reply:
you are right. both solution are as safe, so I'll approve this PR if there's nothing else holding back. I still like it more to separate verifications in `validate` from configuration in `configure`.